src/rr/dnssec/algorithm.rs

```diff
@@ -230,7 +310,15 @@ impl Algorithm {
         bytes.extend_from_slice(&n);
 
         bytes
-      }
+      },
+      Algorithm::ECDSAP256SHA256 | Algorithm::ECDSAP384SHA384 => {
+        let pkey = if let DnssecKey::Ecdsa(ref key) = *public_key {
+            key
+        } else {
+            panic!("Invalid type of key")
+        };
+        pkey.to_wire()
```

Comment:
> In DNSSEC keys, Q is a simple bit string that represents the
>    uncompressed form of a curve point, "x | y".

The format of public keys accepted by *ring* is 0x00 || x || y, where x and y are fixed-length integers. So, basically, add a zero byte in front of what you parsed from the record and that will be the public key.